app/models/blueprint.rb

```diff
@@ -1,4 +1,6 @@
 class Blueprint < ApplicationRecord
+  has_many :versions, :class_name => 'Blueprint',
+           :foreign_key => :original_blueprint_id, :primary_key => :original_blueprint_id
```

Comment:
@jntullo ancestry style is pretty easy.  You add a column named `ancestry` size varchar(1000) and add an index as well.  Then at the top of the model you say `has_ancestry`, and voila you have all the tree-like methods you need.  When you are creating a new node, just pass `:parent => parent_node` and it will do all the relationship updates for you.